lib/geo_redirect/middleware.rb

```diff
@@ -60,9 +61,17 @@ def force_redirect?
 
     def skip_redirect?
       url = URI.parse(@request.url)
+      query_includes_skip_geo?(url) or path_excluded?(url)
+    end
+
+    def query_includes_skip_geo?(url)
       Rack::Utils.parse_query(url.query).key? 'skip_geo'
     end
 
+    def path_excluded?(url)
+      @excludes.reduce(false){ | result, exclude | result || url.path == exclude }
```

Comment:
1. Why not use [#any?](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-any-3F)

2. What do you think about allowing regular expressions here?
So line 18 could be
```ruby
@exclude = Regexp.union(options[:exclude]) if @options[:exclude]
```
and then here in 72 all we need is
```ruby
@exclude && !(@exclude =~ url).nil?
```